doc/man3/OSSL_PROVIDER.pod

```diff
@@ -101,6 +107,14 @@ loaded and initialized or if I<retain_fallbacks> is nonzero.
 If the provider loads successfully and I<retain_fallbacks> is zero, the
 fallback providers are disabled.
 
+OSSL_PROVIDER_load_ex() and OSSL_PROVIDER_try_load_ex() are the variants
+of the previous functions accepting an C<OSSL_PARAM> array of the parameters
+that are passed as the configuration of the loaded provider. The parameters
+of any type but C<OSSL_PARAM_UTF8_STRING> are silently ignored. The parameters
+are added to the ones specified in the configuration file, if any. If config
+parameters differ from the ones provided to the function call, the result is
+not guaranteed.
```

Comment:
I am wondering whether this "additive" behaviour is the correct one. If the parameters from config and the parameters from application code conflict in any way then the result could be unexpected.

I am not entirely sure what the right answer is here. Possible things we could do:

1) application params always take precedence over, and replace params in the config file
2) config file params always taken precedence over, and replace application params
3) application params are additive (as now)...but then what should happen if they both specify the same param? Or the params conflict in some non-obvious way?
4) Trying to specify application params where config params exist causes a failure
5) Some additional "flag" parameter which specifies what to do
6) Provide some mechanism for an application to query the config params??

Reply:
The update is pushed.

Reply:
If you do 5 you need to consider multiple cases of potentially desired behavior:

a) only application provided parameters are considered
b) application parameters are merged to config provided, but applications parameters take precedence.
c) application parameters are merged to config provided, but config parameters take precedence.
d) only config parameters are used if available (ie the passed parameters are only a fallback if there is no config at all).

However if (6) is provided, then no flag needs to be implemented because all can be handled explicitly by the application in constructing the params array, and the behavior would be that the application always overrides the config file when explicitly loading params.
Helper functions to provide a,b,c,d behavior can then be provided outside of the loading mechanism, making the mechanism simpler.